### DISCUSSION TITLE
Invalid HTTP request received, websocket

### DISCUSSION THREAD

**Discussion by Bryan-u-2** [fastapi/fastapi#8564] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import string
from typing import Dict
from fastapi import WebSocket

class ConnectionManager:
    def __init__(self):
        self.active_connections: Dict[str,WebSocket] = {}

    async def connect(self, websocket: WebSocket, userid: str):
        await websocket.accept()
        self.active_connections[userid]=websocket

    def disconnect(self, userid: str):
        del self.active_connections[userid]

    async def broadcast(self, data):
        for k in self.active_connections.keys():
            await (self.active_connections[k]).send_json(data)
            
    async def get_active_users(self):
        return [a for a in self.active_connections.keys()]
```

### Description

Whenver i try to connect to websocket i get  **Invalid HTTP request received.**

### Operating System

Windows

### Operating System Details

win11

### FastAPI Version

0.70.1

### Python Version

Python 3.10.1

### Additional Context

it was working about a year ago,i was just updating the project.

**Comment by chb567** [comment, marked as answer]
Had the same issue on linux with firefox.
I doubt its a problem with fastapi or startlette but appears to be an issue with websockets itself or uvicorn.

Try specifying the "WebSocket protocol implementation" argument `-ws` on uvicorn as wsproto:
`uvicorn main:app --ws  wsproto`

Hope that helps!
